### ISSUE TITLE
Response.map<D: Decodable> fails when using a keyPath pointing to an array

### ISSUE THREAD

**Issue by jgongo** [Moya/Moya#1414]
I'm using `Moya/RxSwift` 10.0.0-beta.1. In this version, when using `map<D: Decodable>` with a keyPath pointing to an array I get an error instead of the decoded array.

If you take a look at the offending code in `Response.swift` you'll see that the result of getting the value at `keyPath` is casted to `[String: Any] `, causing the error when the value is an array instead of a JSON object:

```swift
guard let jsonDictionary = (try mapJSON() as? NSDictionary)?.value(forKeyPath: keyPath) as? [String: Any] else {
    throw MoyaError.jsonMapping(self)
}
do {
    jsonData = try JSONSerialization.data(withJSONObject: jsonDictionary, options: [])
} catch {
    throw MoyaError.jsonMapping(self)
}
```

I think that simply taking out the casting would solve the problem, as any problem with the retrieved value would be captured in the do-catch block just below the code causing the error.

**Comment by SD10** [comment]
@jgongo Thanks for reporting this 👍 Are you talking about #1382? This has been resolved in `Moya 10.0`. We have an official major version now, so you can upgrade from beta 😃

**Comment by jgongo** [comment]
Yes, exactly that. Sorry for the noise! And congratulations for this great framework!
